frontend/javascripts/oxalis/model/accessors/dataset_layer_rotation_accessor.ts

```diff
@@ -0,0 +1,414 @@
+    if (type === "affine") {
+      const nestedMatrix = coordTransformation.matrix;
+      return createAffineTransformFromMatrix(nestedMatrix);
+    } else if (type === "thin_plate_spline") {
+      const { source, target } = coordTransformation.correspondences;
+
+      return createThinPlateSplineTransform(source, target, dataset.dataSource.scale.factor);
+    }
+
+    console.error(
+      "Data layer has defined a coordinate transform that is not affine or thin_plate_spline. This is currently not supported and ignored",
+    );
+    return IdentityTransform;
+  });
+  return transforms.reduce(chainTransforms, IdentityTransform);
+}
+
+export const getOriginalTransformsForLayerOrNull = memoizeWithTwoKeys(
+  _getOriginalTransformsForLayerOrNull,
+);
+
+export function isLayerWithoutTransformationConfigSupport(layer: APIDataLayer | APISkeletonLayer) {
+  return (
+    layer.category === "skeleton" || (layer.category === "segmentation" && !layer.fallbackLayer)
+  );
+}
+
+function _getTransformsForLayerOrNull(
+  dataset: APIDataset,
+  layer: APIDataLayer | APISkeletonLayer,
+  nativelyRenderedLayerName: string | null,
+): Transform | null {
+  if (isLayerWithoutTransformationConfigSupport(layer)) {
+    return getTransformsForLayerWithoutTransformationConfigOrNull(
+      dataset,
+      nativelyRenderedLayerName,
+    );
+  }
+  const layerTransforms = getOriginalTransformsForLayerOrNull(dataset, layer as APIDataLayer);
+
+  if (layer.name === nativelyRenderedLayerName) {
+    // This layer should be rendered without any transforms.
+    return null;
+  }
+  if (nativelyRenderedLayerName == null) {
+    // No layer is requested to be rendered natively. -> We can use the layer's transforms as is.
+    return layerTransforms;
+  }
+
+  // Apply the inverse of the layer that should be rendered natively
+  // to the current layer's transforms.
+  const nativeLayer = getLayerByName(dataset, nativelyRenderedLayerName, true);
+  const transformsOfNativeLayer = getOriginalTransformsForLayerOrNull(dataset, nativeLayer);
+
+  if (transformsOfNativeLayer == null) {
+    // The inverse of no transforms, are no transforms. Leave the layer
+    // transforms untouched.
+    return layerTransforms;
+  }
+
+  const inverseNativeTransforms = invertTransform(transformsOfNativeLayer);
+  return chainTransforms(layerTransforms, inverseNativeTransforms);
+}
+
+export const getTransformsForLayerOrNull = memoizeWithThreeKeys(_getTransformsForLayerOrNull);
+export function getTransformsForLayer(
+  dataset: APIDataset,
+  layer: APIDataLayer | APISkeletonLayer,
+  nativelyRenderedLayerName: string | null,
+): Transform {
+  return (
+    getTransformsForLayerOrNull(dataset, layer, nativelyRenderedLayerName) || IdentityTransform
+  );
+}
+
+export function isIdentityTransform(transform: Transform) {
+  return transform.type === "affine" && _.isEqual(transform.affineMatrix, Identity4x4);
+}
+
+function _getTransformsForLayerWithoutTransformationConfigOrNull(
```

Comment:
at first, I interpreted this as "get transforms for a layer but ignore the transformation config". maybe rename to "_getTransformsForLayerThatDoesNotSupportTransformationConfigOrNull". it's a mouthful, but I'd rather use that to avoid confusion.